File: source/bindings/csharp/resource_handling.cs
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//

using System;

using Microsoft.CognitiveServices.Speech;
using Microsoft.CognitiveServices.Speech.Intent;
using Microsoft.CognitiveServices.Speech.Translation;

namespace Microsoft.CognitiveServices.Speech.Conversation
{
    /// <summary>
    /// Defines how the service handles allocated resources after the conversation transcriber is stopped.
    /// Added in version 1.7.0.
    /// </summary>
    public enum ResourceHandling
    {
        /// <summary>
        /// The service destroys allocated resources.
        /// </summary>
        DestroyResources = 0,

        /// <summary>
        /// The service keeps allocated resources.
        /// </summary>
        KeepResources = 1
    }
}

File: source/bindings/java/com/microsoft/cognitiveservices/speech/conversation/ResourceHandling.java
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//
package com.microsoft.cognitiveservices.speech.conversation;

/**
 * Defines how the service handles allocated resources after the conversation transcriber is stopped.
 * Added in version 1.7.0
 */
public enum ResourceHandling
{
    /**
     * The service destroys allocated resources, such as adapted language models.
     */
    DestroyResources(com.microsoft.cognitiveservices.speech.internal.ResourceHandling.DestroyResources),

    /**
     * The service keeps allocated resources.
     */
    KeepResources(com.microsoft.cognitiveservices.speech.internal.ResourceHandling.KeepResources);

    private ResourceHandling(com.microsoft.cognitiveservices.speech.internal.ResourceHandling id) {
        this.id = id;
    }

    /**
     * Returns the internal ResourceHandling value
     *
     * @return the ResourceHandling value
     */
    public com.microsoft.cognitiveservices.speech.internal.ResourceHandling getValue() {
        return this.id;
    }

    private final com.microsoft.cognitiveservices.speech.internal.ResourceHandling id;

}
